src/tiled/documentmanager.cpp

```diff
@@ -495,7 +495,11 @@ bool DocumentManager::saveDocumentAs(Document *document)
         FormatHelper<MapFormat> helper(FileFormat::ReadWrite);
         filter = helper.filter();
 
-        fileName = getSaveFileName(QCoreApplication::translate("Tiled::Internal::MainWindow", "untitled.tmx"));
+        auto suggestedFileName = QCoreApplication::translate("Tiled::Internal::MainWindow", "untitled");
+        if(!suggestedFileName.isEmpty())
```

Comment:
Coding style: please put a space after `if`

However, in this case you can just remove the condition, since it would be an error for the translated string to be empty (it can never be, because in case of an empty translation, the source string is returned).